### DISCUSSION TITLE
Importing a vm from another source - cloud-init issues

### DISCUSSION THREAD

**Discussion by tdtmusic2** [apache/cloudstack#9846] [Q&A] [answered]
Hi everyone. Did any of you manage to switch to ACS from another similar solution via the import feature? In my case I ended up with a few issues, hopefully someone will point me in the right direction. The current app has a number of vms on kvm hypervisors. The instances are not in qcow2 format, they're img but ACS can import those too. My first issue was with the template being applied to the imported vm, that default or dummy import template - it's not password enabled. I couldn't find a way to apply a different template on the vm other than editing the template id directly in the database, so it would use one that is password enabled (yes, I need that). The second issue is that no matter how I try cloud-init does not work on the imported vm (I apply user-data on them) but it does work on any instance created directly in ACS. The imported vm also uses cloud-init and somehow it seems there's a conflict somewhere, since user-data does not apply. If I look at the output log of cloud-init from that vm it does not seem to do anything, the only line that I see related to cloud-init is about datasource: nocloud and seed: /dev/sda, I have no idea what this is. If I do a cloud-init clean on that vm then user-data from ACS works perfectly but I would really like it to work right after importing the vm, if that's possible.
Anyone else having these issues so far?

Thank you!

**Comment by vishesh92** [comment]
As per the documentation [here](https://docs.cloudstack.apache.org/en/4.19.0.0/adminguide/templates/_cloud_init.html), you need to have this in your cloud-init config for cloud-init to work. Can you try adding this and see if that works for you?
```yaml
datasource_list: [ ConfigDrive, CloudStack, None ]
datasource:
  CloudStack: {}
  None: {}
```

**Reply by tdtmusic2** [reply]
But can I add them as user-data? My user data format is like this, do I add the datasource under #cloud-config and that's it?:
```
#cloud-config
hostname: example
manage_etc_hosts: true
system_info:
  default_user:
    name: user
```

**Reply by weizhouapache** [reply, marked as answer]
> But can I add them as user-data? My user data format is like this, do I add the datasource under #cloud-config and that's it?:
> 
> ```
> #cloud-config
> hostname: example
> manage_etc_hosts: true
> system_info:
>   default_user:
>     name: user
> ```

No, it won't work

cloud-init relies on `datasource_list: [ CloudStack, xxx, xxx]` to get the metadata and user-data from CloudStack VR (acts as the data server). So you must configure CloudStack in the datasource_list before cloud-init starts.
